src/identity.js

```diff
@@ -436,93 +436,77 @@ export class Identity extends EventEmitter {
      * @return {Promise<Identity#HasSessionSuccessResponse|Identity#HasSessionFailureResponse>}
      */
     hasSession(autologin = true) {
-        if (this._hasSessionInProgress) {
-            return this._hasSessionInProgress;
-        }
-        const promiseFn = async (resolve, reject) => {
-            const postProcess = (sessionData) => {
+        if (!this._hasSessionInProgress) {
+            if (typeof autologin !== 'boolean') {
+                const [type, value] = inspect(autologin);
+                return Promise.reject(new SDKError(`Parameter 'autologin' must be boolean, was: "${type}:${value}"`));
+            }
```

Comment:
By doing this validation here rather than in the old Promise executor, we're able to simplify the error handling with a single `catch`-block on [line 501](https://github.com/schibsted/account-sdk-browser/pull/124/files#diff-eaef817870a2a9f52ca5c3938e630207R501)